assets/src/edit-story/components/library/mediaLibrary.js

```diff
@@ -17,50 +17,131 @@ import useLibrary from './useLibrary';
-	width: 150px;
-	padding: 4px;
-	margin: 4px;
+	width: 48%;
+	padding: 3px;
+	margin: 3px;
 	border: 1px solid white;
 `;
 
+const Title = styled.h3`
+	color: ${ ( { theme } ) => theme.colors.fg.v1 };
+	margin: 0px;
+	font-size: 19px;
+	line-height: 20px;
+	line-height: 1.4em;
+	flex: 2 0 0;
+`;
+
+const Button = styled.button`
+	 background: #242A3B;
+	 color: #ffffff;
+	 padding: 5px;
+	 font-weight: bold;
+	 flex: 1 0 0;
+	 text-align: center;
+	 border: 0px none;
+`;
+
+const Header = styled.div`
+	display: flex;
+	margin: 0px 4px 10px;
+`;
+
+const Message = styled.div`
+	color: #ffffff;
+	font-size: 19px;
+`;
+
 function MediaLibrary( { onInsert } ) {
 	const {
-		state: { media, isMediaLoading, isMediaLoaded },
-		actions: { loadMedia },
+		state: { media, isMediaLoading, isMediaLoaded, mediaType },
+		actions: { loadMedia, setIsMediaLoading, setIsMediaLoaded },
 	} = useLibrary();
 
-	useEffect( loadMedia );
+	useEffect( () => {
+		loadMedia();
+		// Work around that forces default tab as upload tab.
+		wp.media.controller.Library.prototype.defaults.contentUserSetting = false;
+	} );
+
+	const mediaPicker = () => {
+		// Create the media frame.
+		const fileFrame = wp.media( {
+			title: 'Upload to Story',
+			button: {
+				text: 'Insert into page',
+			},
+			multiple: false,
+			library: {
+				type: mediaType,
+			},
+		} );
+		let attachment;
+
+		// When an image is selected, run a callback.
+		fileFrame.on( 'select', () => {
+			attachment = fileFrame.state().get( 'selection' ).first().toJSON();
+			const { url } = attachment;
+			onInsert( 'image', {
+				src: url,
+				width: 100,
+				height: 100,
+				x: 5,
+				y: 5,
+				rotationAngle: 0,
+			} );
+		} );
 
-	if ( isMediaLoading || ! isMediaLoaded ) {
-		return <Spinner />;
-	}
+		fileFrame.on( 'close', () => {
+			setIsMediaLoading( false );
+			setIsMediaLoaded( false );
+		} );
 
-	if ( ! media.length ) {
-		return (
-			<div>
-				{ 'No media found' }
-			</div>
-		);
-	}
+		// Finally, open the modal
+		fileFrame.open();
+	};
+
+	const uploadMedia = () => {
+		mediaPicker();
+	};
 
 	return (
 		<div>
-			{ media.map( ( { src } ) => (
-				<Image
-					key={ src }
-					src={ src }
-					width={ 150 }
-					height={ 150 }
-					onClick={ () => onInsert( 'image', {
-						src,
-						width: 200,
-						height: 100,
-						x: 5,
-						y: 5,
-						rotationAngle: 0,
-					} ) }
-				/>
-			) )
+			<Header>
+				<Title>
+					{ 'Media' }
+					{ ( ! isMediaLoaded || isMediaLoading ) &&
+						<Spinner />
+					}
+				</Title>
+				<Button onClick={ uploadMedia }>
+					{ 'Upload' }
+				</Button>
+			</Header>
+
+			{ ( isMediaLoaded && ! media.length ) ? (
+				<Message>
+					{ 'No media found' }
+				</Message>
+			) : (
+
+				media.map( ( { src } ) => (
+					<Image
+						key={ src }
+						src={ src }
+						width={ 150 }
+						height={ 150 }
+						loading={ 'lazy' }
+						onClick={ () => onInsert( 'image', {
+							src,
+							width: 200,
+							height: 100,
```

Comment:
We can leave this improvement for the future but would be great to have the correct aspect ratio when inserting the element.

Reply:
I have fixed this in - https://github.com/ampproject/amp-wp/blob/media-library/assets/src/edit-story/components/library/mediaLibrary.js#L187-L191

Reply:
Nice!